src/main/java/io/github/moulberry/notenoughupdates/listener/RenderListener.java

```diff
@@ -748,6 +755,23 @@ private void renderDungeonChestOverlay(GuiScreen gui) {
 					for (int i = 0; i < 5; i++) {
 						ItemStack item = lower.getStackInSlot(11 + i);
 						String internal = neu.manager.getInternalNameForItem(item);
+						String displayName = item.getDisplayName();
+						if (displayName.contains(" Essence §8x") && neu.config.dungeons.useEssenceCostFromBazaar) {
+							String type = io.github.moulberry.notenoughupdates.core.util.StringUtils.substringBetween(
+								displayName,
+								"§d",
+								" Essence"
+							);
+							JsonObject bazaarInfo = neu.manager.auctionManager.getBazaarInfo("ESSENCE_" + type.toUpperCase());
+							if (bazaarInfo != null && bazaarInfo.has("curr_sell")) {
+								float bazaarPrice = bazaarInfo.get("curr_sell").getAsFloat();
+								int amount = Integer.parseInt(displayName.split(Pattern.quote("§8x"))[1]);
```

Comment:
Again, since you dont regexp match the entire line, i would be careful using Integer.parseInt without exception handling.